pkg/hooks/webhook.go

```diff
@@ -63,27 +73,70 @@ func NewWebhookExecutor(
 		hookType,
 		client,
 		url)
+	if err != nil {
+		return nil, err
+	}
+	return NewWebhookExecutorWithCustomHttpClient(webhook, hookType, client)
+}
+
+func NewWebhookExecutorWithCustomHttpClient(
+	webhook *v1alpha1.Webhook,
+	hookType common.HookType,
+	httpClient HttpClientInterface,
+) (*WebhookExecutor, error) {
+	if webhook == nil {
+		return nil, nil
+	}
+	url, err := webhookURL(webhook)
+	if err != nil {
+		return nil, err
+	}
+	if err != nil {
+		logging.Logger.Info(err.Error())
+	}
+
+	var eTagCache *etag_cache.Cache = nil
+	if webhook.Etag != nil && webhook.Etag.Enabled != nil && *webhook.Etag.Enabled {
+		eTagCache = etag_cache.NewCache(webhook.Etag.CacheTimeoutSeconds, webhook.Etag.CacheTimeoutSeconds)
+	}
+
```

Comment:
should remove below nil check for err since it is checked above

Reply:
thanks, a leftover, removed